db.go

```diff
@@ -566,6 +567,46 @@ func validateBlockSequence(bs []Block) error {
 	return nil
 }
 
+// StartTime return the first timestamp stored in the DB.
+func (db *DB) StartTime() int64 {
+	db.mtx.RLock()
+	defer db.mtx.RUnlock()
+
+	b := db.blocks[0]
+	startTime := int64(math.MaxInt64)
+
+	indexr := b.Index()
+	joblabel := "job"
+	tpls, err := indexr.LabelValues(joblabel)
```

Comment:
That's leaking Prometheus semantics into TSDB, which we generally want to avoid.
I think it's somewhat questionable how well the overall query has something to do with TSDB.

 It's only accessing public interfaces so it doesn't have to be in here.
Anything speaking against implementing this in `prometheus/storage/tsdb`?

Reply:
Could work.

Would you see another way to get the first time without having to do this hack with a label ? Maybe iterating on the chunks directly ? (I'm not super familiar with the interfaces yet)

Reply:
@fabxc It doesn't only access public interfacess. It need access to blocks list which are not public yet (and I thought it was private on purpose)
@iksaif I didn't find an other way. Especially I couldn't access to a chunks list i could iterate on it. 

I only see two possible solutions :
(1) make blocks public adding a BD.Blocks() func (and move the logic to `prometheus/storage/tsdb`)
(2) iterate on existing labels (using block.Index().LabelIndices()) instead of using "job" labels. It means keeping the same logic without prometheus specific labelname.
What do you think ?

Reply:
Making certain methods public if there's a use case for it is perfectly valid. The exported interfaces so far are largely on a best-guess basis :)

Reply:
Done